[SRVCOM-1840] Add Serverless 1.24.0 release notes (#48815)
openshift/openshift-docs: modules/serverless-rn-1-24-0.adoc

// Module included in the following assemblies
//
// * /serverless/serverless-release-notes.adoc

:_content-type: REFERENCE
[id="serverless-rn-1-24-0_{context}"]
= Release notes for Red Hat {ServerlessProductName} 1.24.0

{ServerlessProductName} 1.24.0 is now available. New features, changes, and known issues that pertain to {ServerlessProductName} on {product-title} are included in this topic.

[id="new-features-1.24.0_{context}"]
== New features

* {ServerlessProductName} now uses Knative Serving 1.3.
* {ServerlessProductName} now uses Knative Eventing 1.3.
* {ServerlessProductName} now uses Kourier 1.3.
* {ServerlessProductName} now uses Knative `kn` CLI 1.3.
* {ServerlessProductName} now uses Knative Kafka 1.3.
* The `kn func` CLI plug-in now uses `func` 0.24.

* Init containers support for Knative services is now generally available (GA).

> * FIPS is now disabled for the `KafkaBroker`, `KafkaSource` and `KafkaSink` components.

Reviewer: ```suggestion
* FIPS is now disabled for `KafkaBroker`, `KafkaSource` and `KafkaSink` objects.
```
suggestion per docs guidelines

ifdef::openshift-enterprise[]

> * You can now use the link:serverless/integrations/serverless-cost-management-integration.html#serverless-cost-management-integration_serverless-cost-management-integration[cost management service] with {ServerlessProductName}.

Reviewer: If this link is being included the docs for this will have to be merged before release notes; also it should be an xref, not a link, and therefore shouldn't be included in a module. I would leave the link out entirely.

endif::[]

[id="fixed-issues-1.24.0_{context}"]
== Fixed issues

> * The security issue, where a malicious actor could do a Denial-of-Service (DOS) attack on a server using the `InstrumentHandlerCounter` function in `prometheus/client_golang`, has been fixed. (CVE-2022-21698)

Reviewer 2: There were a few more security issues fixed in this release. If we're mentioning one of them we should maybe list all? Look at [the list](https://issues.redhat.com/issues/?jql=project%20in%20(SRVCLI%2C%20SRVKE%2C%20SRVKS%2C%20SRVCOM%2C%20SRVOCF)%20AND%20issuetype%20%3D%20Bug%20AND%20fixVersion%20%3D%201.24.0)

Author: The set of documented release notes is the set of issues linked to the release notes Jira. Can you folks choose what else needs to be documented and link it to SRVCOM-1840 please?

The blank space at the top appears only in previews, so it's good. 👍

Author: As discussed elsewhere, we have decided to not add any CVE descriptions for 1.24, including this one. Removed.

> * The security issue, where a malicious actor could do a Denial-of-Service (DOS) attack on a server using the `InstrumentHandlerCounter` function in `prometheus/client_golang`, has been fixed. (CVE-2022-21698)

Reviewer: ```suggestion
* In previous versions of {ServerlessProductName}, it was possible for a malicious actor to cause a Denial-of-Service (DOS) attack on a server, by using the `InstrumentHandlerCounter` function in `prometheus/client_golang`. This issue has been fixed (CVE-2022-21698).
```

suggested format for fixed issues

Author: Deleted the CVE description altogether, per discussions.

> * Previously, integrating {ServerlessProductName} with {SMProductName} caused the `net-istio-controller` pod to run out of memory on startup, unless set to have 2Gi memory limit. The bug has been fixed, and `net-istio-controller` no longer requires the increased memory limit.

Reviewer 3: if this refers to https://issues.redhat.com/browse/SRVKS-900 , the text shoud refer to the docs added in https://github.com/openshift/openshift-docs/pull/47036/files  (enabling secret filtering), also I'd consider these changes:

* Previously, integrating {ServerlessProductName} with {SMProductName} caused the `net-istio-controller` pod to run out of memory on startup _when there are too many Secrets present on the cluster_, 
* (2Gi limit is too specific, actual limit may have depended on the number and sizes of the Secrets on the cluster
* Rather than "The bug has been fixed, and `net-istio-controller` no longer requires the increased memory limit.", I think this needs to something along the lines of

>  It is now possible to enable secret filtering, which will make `net-istio-controller` to only consider Secrets with a `networking.internal.knative.dev/certificate-uid` label, thus no longer requiring an increased memory limit.

Reviewer: see above for suggested phrasing for consistency

Author: Fixed accordingly. With Marek's new info, I've removed the "Previously," because the situation persists, it's just that we now have a setting to deal with it.

// Module included in the following assemblies
//
// * /serverless/serverless-release-notes.adoc

:_content-type: REFERENCE
[id="serverless-rn-1-24-0_{context}"]
= Release notes for Red Hat {ServerlessProductName} 1.24.0

{ServerlessProductName} 1.24.0 is now available. New features, changes, and known issues that pertain to {ServerlessProductName} on {product-title} are included in this topic.

[id="new-features-1.24.0_{context}"]
== New features

* {ServerlessProductName} now uses Knative Serving 1.3.
* {ServerlessProductName} now uses Knative Eventing 1.3.
* {ServerlessProductName} now uses Kourier 1.3.
* {ServerlessProductName} now uses Knative `kn` CLI 1.3.
* {ServerlessProductName} now uses Knative Kafka 1.3.
* The `kn func` CLI plug-in now uses `func` 0.24.

* Init containers support for Knative services is now generally available (GA).

> * FIPS is now disabled for `KafkaBroker`, `KafkaSource,` and `KafkaSink` objects.

Reviewer 3: we used to have the FIPS line as a known-issue in 1.23 and 1.22. (I don't think there was any change in relation to FIPS in this release, so it'd be better to keep the same line as a known-issue like in 1.23 and 1.22)

// Module included in the following assemblies
//
// * /serverless/serverless-release-notes.adoc

:_content-type: REFERENCE
[id="serverless-rn-1-24-0_{context}"]
= Release notes for Red Hat {ServerlessProductName} 1.24.0

{ServerlessProductName} 1.24.0 is now available. New features, changes, and known issues that pertain to {ServerlessProductName} on {product-title} are included in this topic.

[id="new-features-1.24.0_{context}"]
== New features

* {ServerlessProductName} now uses Knative Serving 1.3.
* {ServerlessProductName} now uses Knative Eventing 1.3.
* {ServerlessProductName} now uses Kourier 1.3.
* {ServerlessProductName} now uses Knative `kn` CLI 1.3.
* {ServerlessProductName} now uses Knative Kafka 1.3.
* The `kn func` CLI plug-in now uses `func` 0.24.

* Init containers support for Knative services is now generally available (GA).

ifdef::openshift-enterprise[]
* You can now use the cost management service with {ServerlessProductName}.
endif::[]

[id="fixed-issues-1.24.0_{context}"]
== Fixed issues

> * Integrating {ServerlessProductName} with {SMProductName} causes the `net-istio-controller` pod to run out of memory on startup when too many secrets are present on the cluster.
> +
> It is now possible to enable secret filtering, which makes `net-istio-controller` to only consider secrets with a `networking.internal.knative.dev/certificate-uid` label, thus no longer requiring an increased memory limit.

Reviewer: ```suggestion
* In previous versions, integrating {ServerlessProductName} with {SMProductName} caused the `net-istio-controller` pod to run out of memory on startup when too many secrets were present on the cluster. It is now possible to enable secret filtering so that `net-istio-controller` only considers secrets with a `networking.internal.knative.dev/certificate-uid` label. This means that the `net-istio-controller` pod no longer requires an increased memory limit.
```

Author: @abrennan89 Integrating {ServerlessProductName} with {SMProductName} still causes that, so we shouldn't state that this applies only to previous versions. And I prefer to keep the paragraphs separate here, as they are very distinct.

Reviewer: In that case I'd separate them entirely and make the first paragraph a known issue instead

Author: @abrennan89 It's not an issue, it's expected that a big number of secrets require a big amount of memory. It's just now we provide a feature of reducing the number of secrets => reducing amount of required memory. The first paragraph provides the reason for having the feature.
@maschmid Please correct me if I'm wrong.

Author: Resolving this, per Marek's approve. Feel free to reopen if still not agreeing.

Reviewer 3: Both viewpoints make sense IMO,

you can either view it as a bug that it needs to keep all cluster Secrets in memory, with the secret filtering being a kind of workaround;

or you could view the memory requirements as inherent to how the `net-istio-controller` informer works, with the secret filtering being an additional feature to allow you to choose (higher memory limits for `net-istio-controller`, or secret filtering)

FWIW, according to the state of https://issues.redhat.com/browse/SRVKS-900, engineering considers the addition of secret filtering feature as being the fix for the issue. (there's no open known issue for "`net-istio-controller` needs to keep all the cluster secrets in memory")

> It is now possible to enable secret filtering, which makes `net-istio-controller` to only consider secrets with a `networking.internal.knative.dev/certificate-uid` label, thus no longer requiring an increased memory limit.

Reviewer: I'd still suggest:
```
It is now possible to enable secret filtering so that `net-istio-controller` only considers secrets with a `networking.internal.knative.dev/certificate-uid` label. This means that the `net-istio-controller` pod no longer requires an increased memory limit.
```

up to you, though

> It is now possible to enable secret filtering, which makes `net-istio-controller` to only consider secrets with a `networking.internal.knative.dev/certificate-uid` label, thus no longer requiring an increased memory limit.

Author: (Went with Neal's suggestion)
